File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0

        q = deque([root])
       
        cur_sum=0
        while q:
            cur =q.popleft()
            
            if cur.left and cur.left.left is None and cur.left.right is None :
                cur_sum+=cur.left.val

         

           
            if cur.left :
                q.append(cur.left)
            if cur.right:
                q.append(cur.right)
            


        return cur_sum

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */


var sumOfLeftLeaves = function (root) {
  
   let queue=[root]
   let result =0

  
   while (queue.length>0){
      let node =queue.shift()

 
      if(node.left && !node.left.left && !node.left.right){
         result =result +node.left.val
      }


      if(node.left)queue.push(node.left)
      if(node.right)queue.push(node.right)

   }
   return result
}